ddtrace/writer.py

```diff
@@ -155,6 +168,19 @@ def _log_error_status(self, result, result_name):
                       getattr(result, "status", None), getattr(result, "reason", None),
                       getattr(result, "msg", None))
 
+    def _apply_processing_pipeline(self, traces):
```

Comment:
I'd say to add a docstring here to explain briefly how the pipeline works. But what I'm more interested on, is to specify that `traces` is owned by the `AsyncWorker` thread, so it can be freely modified without using a mutex.